engine/src/main/java/com/datatorrent/stram/StreamingContainerAgent.java

```diff
@@ -438,6 +439,9 @@ private OperatorDeployInfo createOperatorDeployInfo(PTOperator oper)
     if (oper.isOperatorStateLess()) {
       ndi.contextAttributes.put(OperatorContext.STATELESS, true);
     }
+    if (deployOpers.get(oper) != null && deployOpers.get(oper) != 0) {
+      ndi.deployGroupId = deployOpers.get(oper);
+    }
```

Comment:
Since the default value is 0 anyway, you can simplify as

if (deployOpers.get(oper) != null) {
       ndi.deployGroupId = deployOpers.get(oper);
}